env/env.go

```diff
@@ -80,6 +80,10 @@ func (v *viperEnv) Set(key string, value interface{}) {
 	v.Viper.Set(key, value)
 }
 
+func (v *viperEnv) IsSet(key string) bool {
+	return v.Viper.IsSet(key)
```

Comment:
There is a slight drawback to this approach - viper.isSet does a viper.find() which is not the fastest and it is already done during unmarshaling. 

what we do right now in main is 
```
config, err := server.NewConfiguration(env.Default())
```

env.Default uses application.yml file, so currently it's our "choice" to say that we want the file created. We can always do env.New(..) instead of default and do not provide configfile there if we are certain we will not need the file.

Ideally I would propose to drop IsSet method and add a check in viperEnv.Load() to check if the file exists and to SetConfigName and SetConfigPath only if it exists. Log proper stuff if it is missing